### PR TITLE
Allow source data path in ingestion CLI to be a file

### PR DESCRIPTION
**Pull request by andrew-uoa**
Currently the ingestion CLI accepts input of a path to a directory to indicate the source data to be ingested. This PR loosens this constraint, allowing the path to refer to a file as well, to support the BIRU profile/use-case where the ingestion is driven by a YAML file, not a directory. This may help with ingesting from archive files etc. in future.

### CODE REVIEW

File: src/ingestion_biru.py [hunk 1]
```diff
@@ -155,7 +154,14 @@ def check_ingest_and_update_status(
         # Ingest the data object
         obj_name_attr = obj_type.lower()
         logger.info("Ingesting %s: %s", obj_type.lower(), obj_name_attr)
-        result = getattr(self.factory, f"ingest_{obj_type}s")([data_obj])
+
```
Comment (LibbyLi667) [inline src/ingestion_biru.py:157]: I don't quite understand the logic behind it. May need to chat.

File: src/ingestion_factory/factory.py
Comment (LibbyLi667) [inline src/ingestion_factory/factory.py:1]: Looking good. Just need updates/small tweaks on the docstrings for IngestionFactory, ingest_dataset and ingest_datafiles (not your problem at all) - a common issue for the current codebase, so might need to refactor all someday.

**Review by LibbyLi667 — COMMENTED** [review]
All look good to me! Need to tweak a little bit in src/profiles/idw/yaml_wrapper.py to make BIRU code work. Change Line44 to:
`with open(fpath, "w", encoding="utf-8") as f:`

File: src/ingestion_factory/factory.py
Comment (andrew-uoa) [reply src/ingestion_factory/factory.py:1]: Good point, I've updated some of these docstrings, and removed a few static analysis disable statements which don't seem to be needed anymore

**Comment by andrew-uoa** [comment]
> All look good to me! Need to tweak a little bit in src/profiles/idw/yaml_wrapper.py to make BIRU code work. Change Line44 to: `with open(fpath, "w", encoding="utf-8") as f:`

Good spotting, I've fixed this now!

**Review by LibbyLi667 — COMMENTED** [review]
Nice fix! Good to be merged

**Pull request merged**